edifact-edi/2.1/modules/ROOT/pages/edifact-edi-connector-examples.adoc

```diff
@@ -0,0 +1,124 @@
+    interchangeIdPartner="partner"
+    interchangeIdQualifierPartner="ZZZ">
+    <edifact:schemas>
+      <edifact:schema value="/edifact/d96a/ORDERS.esl"/>
+      <edifact:schema value="/edifact/d99b/CUSRES.esl"/>
+    </edifact:schemas>
+  </edifact:config>
+  <flow name="edifactmappingFlow">
+     <http:listener doc:name="Listener" config-ref="HTTP_Listener_config" path="/map" />
+     <edifact:read doc:name="Read" config-ref="EDIFACT_EDI_Config"/>
+     <ee:transform doc:name="Transform Message">
+       <ee:message >
+         <ee:set-payload ><![CDATA[%dw 2.0
+output application/java
+---
+{
+    Messages: {
+        D96A: {
+            ORDERS: payload.Messages.D96A.ORDERS
+        }
+    }
+}]]></ee:set-payload>
+       </ee:message>
+     </ee:transform>
+     <edifact:write doc:name="Write" config-ref="EDIFACT_EDI_Config"/>
+  </flow>
+</mule>
+----
+
+== Read and Validate EDIFACT EDI Messages
+
+. To read an EDIFACT message, search the palette for EDIFACT EDI, and drag the EDIFACT Read operation into a flow.
+
+. Go to the properties view, select the connector configuration you created and select the `Read` operation.
+
+This reads any byte stream into the structure described by your EDIFACT schemas.
+
+The EDIFACT module validates the message when it reads it. Message validation includes checking the syntax and content of the envelope segments UNB and UNZ as well as the actual messages within the message. Normally errors are both logged, accumulated, and reported in CONTRL acknowledgment messages and all accepted messages (whether error free or with non-fatal errors) are passed on for processing as part of the output message Map. Exceptions are thrown Errors when reading the input data.
+
+Error data entered in the receive data map uses the `EdifactError` class. This is a read-only JavaBean with these properties.
+
+[%header%autowidth.spread]
+|===
+|Property |Description
+|segment |The zero-based index within the input of the segment causing the error
+|fatal |Flag for a fatal error. This means the associated message, group, or interchange is rejected as a result of the error.
+|errorCode |EDIFACT syntax error code
+|errorText |Text description of the error
+|===
+
+Error data is returned by the read operation as optional lists with the `Errors` key at both the root level of the data structure and at the message level. For the message level, the list contains non-fatal errors encountered during the message parsing. For the root level, the list contains both interchange errors and fatal message errors.
+
+== Write EDIFACT EDI Messages
+
+* To write an outgoing message, search the palette for EDIFACT EDI and drag the EDIFACT Write operation into a flow.
+
+* Construct an outgoing EDIFACT EDI message according to the structure as defined for input to the write operation.
+
+If no fatal errors are found in the write operation the normal flow continues. Otherwise, an exception is thrown that describes the error.
```

Comment:
```suggestion
If no fatal errors are found in the Write operation, the normal flow continues. Otherwise, an exception is thrown that describes the error.
```